src/app/screens/Enable/NostrEnable.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect, useState } from "react";
+import Enable from "~/app/components/Enable";
+import Onboard from "~/app/components/onboard";
+import { useAccount } from "~/app/context/AccountContext";
+import api from "~/common/lib/api";
+import type { OriginData } from "~/types";
+
+type Props = {
+  origin: OriginData;
+};
+
+export default function NostrEnable(props: Props) {
+  const [accountComponent, setAccountComponent] =
+    useState<React.ReactNode | null>(null); // State to store the component to render
+
+  const { account } = useAccount();
+
+  useEffect(() => {
+    async function fetchAccountAndSetComponent() {
+      try {
+        const fetchedAccount = await api.getAccount();
+
+        if (fetchedAccount.nostrEnabled) {
+          setAccountComponent(<Enable origin={props.origin} />);
+        } else {
+          setAccountComponent(<Onboard />);
```

Comment:
is it good to set this as a state or should we do this in the render block depending on the state variables that is set?

Reply:
now we return component directly based on the state variable